### ISSUE TITLE
Bugs in `torch.ao.quantization.fuse_modules`  cause operator fusion failure (`Conv2d` and `BatchNorm2d`)

### ISSUE THREAD

**Issue by AnonymousChaseDreamer** [pytorch/pytorch#112579]
### 🐛 Describe the bug

Hi, when I try to fuse the operators (i.e., `Conv2d` and `BatchNorm2d`), the fusion seems to fail. Fused operators should be `ConvBatchNorm2d` but in fact only `Conv2d` is preserved. `BatchNorm2d` is missing! 

After my preliminary confirmation, I found that there are potential bugs in `ao.quantization.fuse_modules`. It seems that the operator fusion of `BatchNorm2d` has not been implemented yet.

Here is the code to reproduce the instance
```python
from torch.ao.quantization.fuse_modules import fuse_modules
import torch.nn as nn


class M(nn.Module):
    def __init__(self):
        super(M, self).__init__()
        self.conv1 = nn.Conv2d(3, 64, 3, 1, 1)
        self.bn1 = nn.BatchNorm2d(64)
        self.conv2 = nn.Conv2d(64, 64, 3, 1, 1)
        self.bn2 = nn.BatchNorm2d(64)

    def forward(self, x):
        x = self.conv1(x)
        x = self.bn1(x)
        x = self.conv2(x)
        x = self.bn2(x)
        return x


m = M().eval()

modules_to_fuse = ['conv1', 'bn1']

print(f'before operator fusion: \n{m}\n')
fused_m = fuse_modules(m, modules_to_fuse)
print(f'after operator fusion: \n{fused_m}\n')
```
Here is the output
```
before operator fusion: 
M(
  (conv1): Conv2d(3, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (bn1): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
  (conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (bn2): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
)

after operator fusion: 
M(
  (conv1): Conv2d(3, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (bn1): Identity()
  (conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (bn2): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
)
```

This is just a minimal instance. In real application, I found that applying operator fusion may result in inconsistency with the original result.

Can you help me confirm this bug? Thx! 

### Versions

Collecting environment information...
PyTorch version: 2.1.0
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.8.2
Libc version: glibc-2.27

Python version: 3.10.13 (main, Sep 11 2023, 13:44:35) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-4.15.0-213-generic-x86_64-with-glibc2.27
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: 
GPU 0: Tesla V100-SXM2-32GB
GPU 1: Tesla V100-SXM2-32GB
GPU 2: Tesla V100-SXM2-32GB
GPU 3: Tesla V100-SXM2-32GB

Nvidia driver version: 470.182.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.24.3
[pip3] onnx==1.13.1
[pip3] onnxoptimizer==0.3.13
[pip3] onnxruntime==1.16.1
[pip3] torch==2.1.0
[pip3] torchaudio==2.1.0
[pip3] torchvision==0.16.0
[conda] Could not collect

**Comment by AnonymousChaseDreamer** [comment]
cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel 
Thx for your help if any advice

**Comment by andrewor14** [comment]
@jerryzh168 Is this a real issue then? We don't do conv-bn fusion for PTQ in eager mode, right? Can we close this?

**Comment by AnonymousChaseDreamer** [comment]
However, the your official document about fuse_modules says you have implemented conv-bn fusion. I feel confused about this

**Comment by AnonymousChaseDreamer** [comment]
In theory, the output after fusion should be the same as before fusion. But this is not true in real applications. I can provide some results for reference if you want

**Comment by jerryzh168** [comment]
oh this is just what post training conv - bn fusion does actually.. it will just fold the weight of bn to conv

**Comment by jerryzh168** [comment]
> @jerryzh168 Is this a real issue then? We don't do conv-bn fusion for PTQ in eager mode, right? Can we close this?

conv-bn fusion for PTQ in eager mode is done explicitly by calling fuse_modules API. yeah I feel we can close this since there is nothing to fix. but maybe we can wait until @AnonymousChaseDreamer got all the questions answered
